### ISSUE TITLE
Windows 11 OpenRGB 0.9 causes the app to crash and stop immediately

### ISSUE THREAD

**Issue by Booma1002** [krojew/OpenRGB-Ambient#16]
Windows 11 OpenRGB 0.9 causes the app to crash and stop immediately.
The other attempts to use older versions of OpenRGB-Ambient were not successful; as the plugin doesn't appear in the app ribbon.

**Comment by Booma1002** [comment]
Yes, I’m sure. It's worth mentioning that I couldn’t test the older versions on OpenRGB 0.8 because certain internal keyboards, including mine, were undetectable in that version. This might be a specific issue with my keyboard model, a "TUF FX506HE ASUS laptop." I would greatly appreciate any assistance you could provide in implementing this feature.

**Comment by krojew** [comment]
Are you able to use a debugger and try to get a stacktrace?

**Comment by Booma1002** [comment]
I have Qt Creator, and some shells like MinGW and MSYS. I cloned the OpenRGB repository using:
i did : 
```
git clone https://gitlab.com/CalcProgrammer1/OpenRGB.git
cd OpenRGB
```
However, when I attempt to debug it in Qt Creator, I encounter errors. If I resolve these errors, I should be able to add the plugin and provide a proper stack trace. Could you guide me through this process?

![Screenshot 2024-08-27 192135](https://github.com/user-attachments/assets/a547059b-cdb7-4f8b-8041-a4c7b347968d)

**Comment by krojew** [comment]
Unfortunately, I can't since I don't use Qt creator. Maybe look for build instructions in their gitlab for that.

**Comment by Booma1002** [comment]
Thanks for the suggestion. I will look for the build instructions on the GitLab repository. If you have any specific recommendations or additional resources for debugging without Qt Creator, please let me know.

**Comment by krojew** [comment]
My way is to use CLion and simply launch a debugging session from there.

**Comment by Booma1002** [comment]
I made a debugging session for the app execution built in Release mode.  Below is the stack trace and related information:
`12:18:40: Debugging C:\Users\Booma\OpenRGB\build\Desktop_Qt_5_15_17_MSVC2019_64bit-Release\release\OpenRGB.exe ...`

`Exception at 0x7fff7283567c, code: 0xe06d7363: C++ exception, flags=0x81 (first chance) in MessageBus!messageBusRemoveObserver`

`Exception at 0x7fff7283567c, code: 0xe06d7363: C++ exception, flags=0x81 (first chance) in 
MessageBus!messageBusRemoveObserver`
`clientcore\windows\advcore\ctf\uim\tim.cpp(800)\MSCTF.dll!00007FFF74AA6709: (caller: 00007FFF74AA733C) LogHr(1) tid(5ee8) 8007029C An assertion failure has occurred.`
`clientcore\windows\advcore\ctf\uim\tim.cpp(800)\MSCTF.dll!00007FFF74AA6709: (caller: 00007FFF74AA733C) LogHr(2) tid(5ee8) 8007029C An assertion failure has occurred.`

`Exception at 0x7fff7283567c, code: 0xe06d7363: C++ exception, flags=0x81 (first chance) in openrgb_ambient__1_`

`1  abort                      ucrtbase        0x7fff7252f61e `
`2  terminate                  ucrtbase        0x7fff7252ed39 `
`3  VCRUNTIME140_1                             0x7fff526f1911 `
`4  VCRUNTIME140_1                             0x7fff526f218f `
`5  VCRUNTIME140_1                             0x7fff526f21e9 `
`6  _CxxFrameHandler4          VCRUNTIME140_1  0x7fff526f4019 `
`7  _chkstk                    ntdll           0x7fff7503441f `
`8  RtlFindCharInUnicodeString ntdll           0x7fff74fae466 `
`9  KiUserExceptionDispatcher  ntdll           0x7fff7503340e `
`10 RaiseException             KERNELBASE      0x7fff7283567c `
`11 CxxThrowException          VCRUNTIME140    0x7fff524c5267` 
`12 openrgb_ambient__1_                        0x7fff30cbbe9c `
`13 openrgb_ambient__1_                        0x7fff30cbb097 `
`14 openrgb_ambient__1_                        0x7fff30cb169a `
`15 recalloc                   ucrtbase        0x7fff724d9363 `
`16 BaseThreadInitThunk        KERNEL32        0x7fff73bd257d `
`17 RtlUserThreadStart         ntdll           0x7fff74feaa58 `

Thank you very much for your time and assistance with this issue. Your expertise and support are invaluable, and I am grateful for your help in addressing these concerns. 🫶

Please let me know if you require any additional information or if there are any further steps I should take.

**Comment by krojew** [comment]
Are you able to jump to point 12 on this list and check what's the file name and line number there?

**Comment by Booma1002** [comment]
File Name: Disassembler (openrgb_ambient__1_)
Line Number: 81
Please note that the plugin i installed is named "openrgb_ambient (1).dll".

**Comment by krojew** [comment]
Ah, you need the debug version of the plugin...

**Comment by Booma1002** [comment]
I have downloaded the source files for the plugin. Could you please clarify the exact steps to build the debug version of the plugin? Specifically, I would like to know:

_**1- If there are any specific build instructions or configurations needed for generating a debug version.
2- How to ensure that the debug symbols are correctly included in the build.**_

**Comment by krojew** [comment]
You just need to select the debug version in your IDE. I don't know which one you're using.

**Comment by Booma1002** [comment]
I was running OpenRGB in Release mode using Qt Creator.
Could you guide me on how to set up and run both the OpenRGB app and plugin in Debug mode using CLion? Specifically:

How do I configure CLion to build and run both in Debug mode?
Are there any CMake settings I need to adjust for debugging the plugin with OpenRGB?
Any tips or common issues to be aware of?

**Comment by krojew** [comment]
You should have a dropdown at the top toolbar with release or debug configuration to choose from.
